### ISSUE TITLE
New regression with some shapes in 0.17.3

### ISSUE THREAD

**Issue by adrian17** [nical/lyon#627]
I confirmed that this regressed in https://github.com/nical/lyon/commit/537104590d13148bdda70b36a45dc6688073f3a0 .

Before/after:
![image](https://user-images.githubusercontent.com/4729533/105089342-48a3ed00-5a9d-11eb-800c-6019bed9718b.png)

Unfortunately the exported SVG has more stuff this time, but most paths are irrelevant.
```svg
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns:xlink="http://www.w3.org/1999/xlink" height="172.9px" width="182.2px" xmlns="http://www.w3.org/2000/svg">
  <g transform="matrix(1.0, 0.0, 0.0, 1.0, 91.1, 86.45)">
    <path d="M78.3 -77.65 L78.65 -78.65 78.7 -78.65 78.8 -78.6 78.3 -77.65" fill="url(#gradient0)" fill-rule="evenodd" stroke="none"/>
    <path d="M-36.9 -86.3 L-35.85 -86.25 -33.6 -86.2 Q22.25 -84.75 76.7 -79.15 L77.05 -79.15 78.65 -78.65 78.3 -77.65 61.4 -62.95 Q-1.75 -69.0 -34.75 -70.5 L-51.35 -71.25 -56.95 -71.45 -51.35 -71.25 -34.75 -70.5 Q-1.75 -69.0 61.4 -62.95 L40.3 -44.2 -80.4 -52.9 -56.95 -71.45 -38.15 -86.15 -36.9 -86.3" fill="#dde1e1" fill-rule="evenodd" stroke="none"/>
    <path d="M-87.7 -47.3 L-87.95 -47.25 -87.1 -48.35 -87.7 -47.3" fill="url(#gradient1)" fill-rule="evenodd" stroke="none"/>
    <path d="M-90.35 -42.7 L-90.25 -43.1 -90.15 -43.05 -90.35 -42.7" fill="url(#gradient2)" fill-rule="evenodd" stroke="none"/>
    <path d="M-85.1 -50.35 L-87.7 -47.3 -87.1 -48.35 -85.1 -50.35" fill="#d3ccc5" fill-rule="evenodd" stroke="none"/>
    <path d="M-36.9 -86.3 L-38.15 -86.15 -38.15 -86.4 -38.1 -86.4 -36.9 -86.3" fill="url(#gradient3)" fill-rule="evenodd" stroke="none"/>
    <path d="M21.35 37.3 L-3.7 35.1 -3.15 58.8 -37.15 55.4 -38.15 32.05 -61.85 30.0 -62.65 -3.05 -39.6 -1.4 -40.65 -26.15 -5.15 -23.65 -4.55 1.05 20.85 2.85 21.35 37.3" fill="#891c10" fill-rule="evenodd" stroke="none"/>
    <path d="M50.7 84.95 L47.85 85.75 50.3 83.85 50.4 83.8 50.7 84.95" fill="url(#gradient4)" fill-rule="evenodd" stroke="none"/>
    <path d="M50.3 83.85 Q43.9 87.4 35.85 86.1 L-76.2 77.45 Q-83.75 75.5 -86.35 66.35 L-90.95 -29.65 -91.05 -33.95 -90.65 -41.4 -90.35 -42.7 Q-90.35 -42.3 -90.25 -42.7 L-90.2 -42.95 -90.15 -43.0 -90.15 -43.05 -89.85 -43.65 -89.1 -45.05 -87.95 -47.25 -87.7 -47.3 -85.1 -50.35 Q-83.5 -51.65 -81.5 -52.5 L-80.4 -52.9 40.3 -44.2 45.5 -43.85 Q55.9 -42.5 58.8 -32.55 59.65 -29.5 59.85 -25.65 L59.85 -24.9 59.85 -24.7 59.85 -24.5 59.85 -24.45 59.85 -24.15 59.85 55.85 Q59.9 60.8 59.4 65.0 L59.35 65.0 59.4 65.1 Q57.7 78.3 50.4 83.8 L50.3 83.85 M21.35 37.3 L20.85 2.85 -4.55 1.05 -5.15 -23.65 -40.65 -26.15 -39.6 -1.4 -62.65 -3.05 -61.85 30.0 -38.15 32.05 -37.15 55.4 -3.15 58.8 -3.7 35.1 21.35 37.3" fill="#c1c8c8" fill-rule="evenodd" stroke="none"/>
    <path d="M90.75 -64.7 Q91.35 -60.95 90.9 -56.3 L90.9 32.85 90.8 37.4 77.95 49.7 59.4 65.0 Q59.9 60.8 59.85 55.85 L59.85 -24.15 59.85 -24.45 59.85 -24.5 59.85 -24.7 59.85 -24.9 59.85 -25.65 Q59.65 -29.5 58.8 -32.55 L59.3 -32.65 76.85 -50.75 90.75 -64.7 M76.85 -50.75 L77.95 49.7 76.85 -50.75" fill="#a9b2b2" fill-rule="evenodd" stroke="none"/>
    <path d="M90.8 37.4 Q90.05 54.0 85.3 58.35 81.25 63.25 77.4 67.2 L77.2 67.4 77.05 67.55 75.0 69.6 Q62.45 81.15 50.7 84.95 L50.4 83.8 Q57.7 78.3 59.4 65.1 L59.35 65.0 59.4 65.0 77.95 49.7 78.05 55.95 78.1 58.1 Q78.1 62.8 76.4 66.85 L75.0 69.6 76.4 66.85 Q78.1 62.8 78.1 58.1 L78.05 55.95 77.95 49.7 90.8 37.4" fill="#9da6a6" fill-rule="evenodd" stroke="none"/>
    <path d="M61.4 -62.95 L78.3 -77.65 78.8 -78.6 Q88.95 -75.1 90.75 -64.7 L76.85 -50.75 59.3 -32.65 58.8 -32.55 Q55.9 -42.5 45.5 -43.85 L40.3 -44.2 61.4 -62.95 69.55 -62.15 Q76.05 -60.9 76.8 -55.65 L76.85 -50.75 76.8 -55.65 Q76.05 -60.9 69.55 -62.15 L61.4 -62.95" fill="#cfdbdb" fill-rule="evenodd" stroke="none"/>
    <path d="M-38.15 -86.15 L-56.95 -71.45 -60.1 -71.6 -66.6 -71.4 -52.4 -82.15 Q-46.45 -86.8 -38.15 -86.4 L-38.15 -86.15 M-90.25 -43.1 L-88.9 -49.25 Q-88.0 -52.25 -85.45 -55.9 L-85.2 -56.25 -83.9 -58.05 -69.8 -70.25 Q-68.25 -71.1 -66.6 -71.4 L-60.1 -71.6 -56.95 -71.45 -80.4 -52.9 -81.5 -52.5 Q-83.5 -51.65 -85.1 -50.35 L-87.1 -48.35 -87.95 -47.25 -89.1 -45.05 -89.85 -43.65 -90.15 -43.05 -90.25 -43.1" fill="#c7cdcd" fill-rule="evenodd" stroke="none"/>
    <path d="M-56.95 -71.45 L-51.35 -71.25 -34.75 -70.5 Q-1.75 -69.0 61.4 -62.95 L69.55 -62.15 Q76.05 -60.9 76.8 -55.65 L76.85 -50.75 M-69.8 -70.25 Q-68.25 -71.1 -66.6 -71.4 M77.95 49.7 L76.85 -50.75 M77.95 49.7 L78.05 55.95 78.1 58.1 Q78.1 62.8 76.4 66.85 L75.0 69.6 M-56.95 -71.45 L-60.1 -71.6 -66.6 -71.4" fill="none" stroke="#666666" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.0"/>
  </g>
  <defs>
    <linearGradient gradientTransform="matrix(0.0084, 0.0084, -0.011, 0.011, 77.3, -65.0)" gradientUnits="userSpaceOnUse" id="gradient0" spreadMethod="pad" x1="-819.2" x2="819.2">
      <stop offset="0.0" stop-color="#d7dbdb"/>
      <stop offset="0.5019607843137255" stop-color="#e4e7e7"/>
      <stop offset="1.0" stop-color="#bec5c5"/>
    </linearGradient>
    <linearGradient gradientTransform="matrix(0.0893, 0.0746, -0.0686, 0.082, -16.0, 15.3)" gradientUnits="userSpaceOnUse" id="gradient1" spreadMethod="pad" x1="-819.2" x2="819.2">
      <stop offset="0.01568627450980392" stop-color="#bec5c5"/>
      <stop offset="1.0" stop-color="#dde1e1"/>
    </linearGradient>
    <linearGradient gradientTransform="matrix(-0.0042, -0.0059, 0.0215, -0.0154, -71.45, -62.1)" gradientUnits="userSpaceOnUse" id="gradient2" spreadMethod="pad" x1="-819.2" x2="819.2">
      <stop offset="0.0" stop-color="#d7dbdb"/>
      <stop offset="0.5019607843137255" stop-color="#e4e7e7"/>
      <stop offset="1.0" stop-color="#bec5c5"/>
    </linearGradient>
    <linearGradient gradientTransform="matrix(-0.0033, -0.0049, 0.0181, -0.012, -49.65, -78.15)" gradientUnits="userSpaceOnUse" id="gradient3" spreadMethod="pad" x1="-819.2" x2="819.2">
      <stop offset="0.0" stop-color="#d7dbdb"/>
      <stop offset="0.5019607843137255" stop-color="#e4e7e7"/>
      <stop offset="1.0" stop-color="#bec5c5"/>
    </linearGradient>
    <linearGradient gradientTransform="matrix(0.0053, 0.0065, -0.0219, 0.0178, 64.45, 69.55)" gradientUnits="userSpaceOnUse" id="gradient4" spreadMethod="pad" x1="-819.2" x2="819.2">
      <stop offset="0.0" stop-color="#bec5c5"/>
      <stop offset="0.4823529411764706" stop-color="#bcc2c2"/>
      <stop offset="1.0" stop-color="#b0b9b9"/>
    </linearGradient>
  </defs>
</svg>
```

**Comment by nical** [comment]
Thanks for filing this. The testcase is very helpful, I could reduce it down to:

```
"M 50.3 83.85 43.9 87.4 35.85 86.1 -83.75 75.5 -90.2 -42.95 59.85 -24.9 59.9 60.8 59.35 65.0 59.4 65.1
M 21.35 37.3 20.85 2.85 -37.15 55.4"
```

I'll  have a look shortly.

**Comment by nical** [comment]
OK I figured it out it's a very silly copy-paste error. I need to figure out a way to better test this part of the code.

**Comment by nical** [comment]
Fixed in Version 0.17.4
